includes/admin/home/tasks/task/class-sensei-home-task-create-first-course.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * File containing the Sensei_Home_Task_Create_First_Course class.
+ *
+ * @package sensei-lms
+ * @since $$next-version$$
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei_Home_Task_Create_First_Course class.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_Home_Task_Create_First_Course implements Sensei_Home_Task {
+	/**
+	 * The ID for the task.
+	 *
+	 * @return string
+	 */
+	public static function get_id(): string {
+		return 'create-first-course';
+	}
+
+	/**
+	 * Number used to sort in frontend.
+	 *
+	 * @return int
+	 */
+	public function get_priority(): int {
+		return 200;
+	}
+
+	/**
+	 * Task title.
+	 *
+	 * @return string
+	 */
+	public function get_title(): string {
+		return __( 'Create your first Course', 'sensei-lms' );
+	}
+
+	/**
+	 * Task url.
+	 *
+	 * @return string
+	 */
+	public function get_url(): ?string {
+		return admin_url( 'post-new.php?post_type=course' );
+	}
+
+	/**
+	 * Task image.
+	 *
+	 * @return string|null
+	 */
+	public function get_image(): ?string {
+		return null;
+	}
+
+	/**
+	 * Whether the task is completed or not.
+	 *
+	 * @return bool
+	 */
+	public function is_completed(): bool {
+		global $wpdb;
+		$cache_key   = 'home/tasks/create-first-course';
+		$cache_group = 'sensei/temporary';
+		$result      = wp_cache_get( $cache_key, $cache_group );
+		if ( false === $result ) {
+			$prefix = $wpdb->esc_like( Sensei_Data_Port_Manager::SAMPLE_COURSE_SLUG );
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery -- Safe-ish and rare query.
+			$result = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM {$wpdb->posts} WHERE post_type='course' AND post_name NOT LIKE %s", "{$prefix}%" ) );
```

Comment:
@fjorgemota Sorry for the late review, but I also wonder if we should do some post status filtering here (mainly to not include trashed posts).

Reply:
Hey, good catch! I created the issue #5934  to fix that. And I'll implement the fix on Sensei Pro too, as the PR there is still open. Thanks!